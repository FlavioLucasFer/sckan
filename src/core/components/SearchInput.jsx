import React from 'react';
import PropTypes from 'prop-types';

SearchInput.propTypes = {
	onClear: PropTypes.func,
};

function SearchInput(props) {
	const { 
		onClear, 
		...rest
	} = props;

	return (
		<div className="nav-wrapper grey lighten-5"
			data-testid="search-bar"
			style={{ borderRadius: '5px', lineHeight: '50px' }}>
			<form onSubmit={e => e.preventDefault()}>
				<div className="input-field">
					<input {...rest}
						className="grey lighten-5"
						id="search" 
						type="search" />
					<label className="label-icon vertical-center-content" htmlFor="search">
						<i className="material-icons-round">
							search
						</i>
					</label>
					<i className="material-icons vertical-center-content"
						onClick={onClear}>
						cancel
					</i>
				</div>
			</form>
		</div>
	);
}

export default SearchInput;