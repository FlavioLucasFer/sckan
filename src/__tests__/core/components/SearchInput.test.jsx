import React from 'react';
import { shallow } from 'enzyme';

import SearchInput from 'core/components/SearchInput';

describe('Testing SearchInput core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<SearchInput />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should change value on change action', () => {
		const wrapper = shallow(
			<SearchInput onChange={handleChange} />
		);

		const event = { target: { value: 'test' } };
		let value = ''; 

		function handleChange(v) {
			value = v;
		}

		wrapper.find('input').simulate('change', event);


		expect(value).toBe('test');
		expect(wrapper.find('input').props().value).toBe('test');
	});

	it('should clear input value when click on cancel icon', () => {
		const wrapper = shallow(
			<SearchInput />
		);

		const event = { target: { value: 'testing' } };
	
		wrapper.find('input').simulate('change', event);
		
		expect(wrapper.find('input').props().value).toBe('testing');
		
		wrapper.find({ children: 'cancel' }).simulate('click');
		
		expect(wrapper.find('input').props().value).toBe('');
	});
});