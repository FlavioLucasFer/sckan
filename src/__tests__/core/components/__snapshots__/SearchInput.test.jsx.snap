// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SearchInput core component should render correctly 1`] = `
"<div className=\\"nav-wrapper grey lighten-5\\" data-testid=\\"search-bar\\" style={{...}}>
  <form onSubmit={[Function: onSubmit]}>
    <div className=\\"input-field\\">
      <input className=\\"grey lighten-5\\" id=\\"search\\" type=\\"search\\" value=\\"\\" onChange={[Function: handleChange]} />
      <label className=\\"label-icon vertical-center-content\\" htmlFor=\\"search\\">
        <i className=\\"material-icons-round\\">
          search
        </i>
      </label>
      <i className=\\"material-icons vertical-center-content\\" onClick={[Function: handleClear]}>
        cancel
      </i>
    </div>
  </form>
</div>"
`;
